### PR TITLE
Added "ItemContainer" class so that items and their quantities can be displayed instead of duplicates

### LINKED ISSUE #12: duplicate items are not shown as Sword x2
duplicate items in inventory, containers, rooms are not shown as Sword x2

### PR DESCRIPTION
**Pull request by sb89**
Fixes #12 

Let me know what you think.

### CODE REVIEW

**Comment by LiamKenneth** [comment]
Brilliant, thanks! This looks really good. I'll check it out and merge it later when I have more time. If you're interested in helping out more / sharing ideas. I have a discord channel for the project: https://discord.gg/nuf7FVq  Would be great to work with someone else on AQ.

File: MIMWebClient/Core/Events/LoadRoom.cs [hunk 1]
```diff
@@ -413,12 +413,9 @@ public static void ReturnRoom(Player player, Room room, string commandOptions =
                                 HubContext.SendToClient("You look into the " + itemDescription.name + " and see:", player.HubGuid);
 
 
-                                foreach (var containerItem in itemDescription.containerItems)
+                                foreach (var containerItem in itemDescription.containerItems.List())
```
Comment (LiamKenneth) [inline MIMWebClient/Core/Events/LoadRoom.cs]: Pass true in the itemcontainer list so we get the article showing before the item.

File: MIMWebClient/Core/Events/LoadRoom.cs [hunk 2]
```diff
@@ -554,12 +551,9 @@ public static void ReturnRoom(Player player, Room room, string commandOptions =
                             {
                                 HubContext.SendToClient("You look into the " + playerItemDescription.name + " and see:", player.HubGuid);
 
-                                foreach (var containerItem in playerItemDescription.containerItems)
+                                foreach (var containerItem in playerItemDescription.containerItems.List())
```
Comment (LiamKenneth) [inline MIMWebClient/Core/Events/LoadRoom.cs]: Pass true in the itemcontainer list so we get the article showing before the item.

File: MIMWebClient/Core/Events/Score.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ public static void UpdateUiInventory(Player playerData)
 
 
                 context.Clients.Client(playerData.HubGuid)
-                    .updateInventory(playerData.Inventory.Where(x => x.location == Item.Item.ItemLocation.Inventory && x.type != Item.Item.ItemType.Gold));
+                    .updateInventory(ItemContainer.List(playerData.Inventory.Where(x => x.location == Item.Item.ItemLocation.Inventory && x.type != Item.Item.ItemType.Gold)));
```
Comment (LiamKenneth) [inline MIMWebClient/Core/Events/Score.cs]: Pass true in the itemcontainer list so we get the article showing before the item.

File: MIMWebClient/Core/ItemContainer.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                Name = group.Key,
+                Count = group.Count()
+            }).Select(x =>
+            {
+                var itemString = x.Name + ((x.Count > 1) ? " x" + x.Count : "");
+                if (article)
+                {
+                    var result = AvsAnLib.AvsAn.Query(x.Name);
+                    itemString = result.Article + " " + itemString;
```
Comment (LiamKenneth) [inline MIMWebClient/Core/ItemContainer.cs]: result.Article should start with a capital letter. I have a helper method here: [Helpers.FirstLetterToUpper(string)](https://github.com/LiamKenneth/ArchaicQuest/blob/b46dd24133fca8529f4c07e79b138848cbd4f896/MIMWebClient/Core/Helpers.cs#L178)

File: MIMWebClient/Core/Player/Inventory.cs [hunk 5]
```diff
@@ -12,26 +12,19 @@ namespace MIMWebClient.Core.Player
...
+            if (inventory != null && inventory.Count > 0)
+            {
+
+                var itemList = new StringBuilder();
+                itemList.Append("You are carrying:").AppendLine();
 
-                    inventoryItems.Append(article.ToUpper() + " " + item.name).AppendLine();
-                }
-                
+                foreach(var item in ItemContainer.List(inventory.Where(x => x.location == Item.ItemLocation.Inventory))){
```
Comment (LiamKenneth) [inline MIMWebClient/Core/Player/Inventory.cs]: Pass true in the itemcontainer list so we get the article showing before the item.

**Comment by sb89** [comment]
I'm having a look at the code in LoadRoom, and it takes the item count into consideration when printing the message. Should the code that I've added also do that? I guess we can add a parameter to configure that?

**Comment by LiamKenneth** [comment]
I noticed that too, It's something I started but never fully completed as it doesn't work. 😊 So was wondering how I could use your code as it works well for containers and inventory.  If not,  shouldn't be too difficult to get the load room items to display the same way. I can have a look at that one.

**Comment by sb89** [comment]
I think It could be written as (if we are not worrying about the item count):

            var items = room.items.Where(x => x.itemFlags?.Contains(Item.Item.ItemFlags.invis) != true && player.DetectInvis || x.itemFlags?.Contains(Item.Item.ItemFlags.hidden) != true && player.DetectHidden);
            items = items.Where(x => !x.isHiddenInRoom);

            foreach (var item in items.Where(x => !string.IsNullOrEmpty(x.description?.room)))
            {
                itemList += $"<p class='roomItems'>{item.description.room}<p>";
            }

            foreach (var item in ItemContainer.List(items.Where(x => string.IsNullOrEmpty(x.description?.room)), true))
            {
                itemList += $"<p class='roomItems'>{item} is on the floor here.<p>";
            }

**Comment by LiamKenneth** [comment]
Thanks for the example, I'll try it out when I get back home.  what do you plan to do next on the project? My issue list is quite out of date. My next steps is to add more skills and spells.  Similar to your sandard diku ones if your familiar with diku muds. Fight code also needs a look into as goblins don't seem to fightback until you've hit them ten times. 

What do you think about the project by the way?

**Pull request merged**